Scripts/RestorePackages.bat

```diff
@@ -3,6 +3,10 @@ CALL %~dp0\InitializeEnvironment.bat || EXIT /b 10
 
 SETLOCAL
 
+IF "%1"=="" (SET "Configuration=Debug") ELSE (SET "Configuration=%1")

```

Comment:
I think this change looks good, but out of curiosity does the configuration matter when restoring nuget packages?  Or could we always restore the Release configuration?

Reply:
When I change the call in RestorePackages.bat to be this:
`dotnet restore %VFS_SRCDIR%\GVFS.sln /p:Configuration=%SolutionConfiguration% /p:VCTargetsPath="C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140" --packages %VFS_PACKAGESDIR% || exit /b 1`

I get this error:
> 1>C:\Users\nickgra\VFSForGit\GVFS.sln.metaproj : error MSB4126: The specified solution configuration "Debug|x64" is invalid. Please specify a valid solution configuration using the Configuration and Platform properties (e.g. MSBuild.exe Solution.sln /p:Configuration=Debug /p:Platform="Any CPU") or leave those properties blank to use the default solution configuration. [C:\Users\nickgra\VFSForGit\GVFS.sln]
1>C:\Users\nickgra\VFSForGit\GVFS\GVFS.Build\GVFS.PreBuild.csproj(185,5): error MSB3073: The command "call C:\Users\nickgra\VFSForGit\GVFS\GVFS.Build\..\..\Scripts\RestorePackages.bat Debug" exited with code 1.
3>------ Build started: Project: GVFS.GVFlt, Configuration: Debug x64 ------

This makes sense according to the docs, nothing implies a default of Debug and we don't have Debug as a configuration in our project.

The configuration doesn't matter here for what packages get restored (at least in our projects), but the Configuration does matter for what asset/metadata files get generated in BuildOutput\<Project Name>\obj\[Platform]\[Configuration]. We have to restore for whatever configuration we're building, we could restore both in this script (which would add more time to first time builds) or we could pass the platform in.

This entire script can be removed whenever we move to .NET Core completely so I think this is good enough for now.